### PR TITLE
Added VRF to interface bulk edit form.

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #4708
# What's Changed

I refactored to combine the code paths for the different ways of accessing the interface bulk edit view

Previously if you wanted to bulk edit the parent interface, bridge interface or lag interface fields you had to go to the DeviceInterfacesView and perform the bulk edit there. With these changes, if you attempt to bulk edit interfaces that all belong to the same device from any view, you will be able to edit those fields. 

This also moves the count of distinct `device__location` from python to the database.

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by HanlinMiao — APPROVED** [review]
Do you mind including some VRF data in the Bulk Edit Test for interfaces?

File: nautobot/dcim/forms.py [hunk 1]
```diff
@@ -2414,42 +2414,43 @@ class Meta:
...
+            .only("id", "device", "device__location")
+        )
+        devices = Device.objects.order_by().filter(interfaces__in=interfaces).only("pk").distinct()
+        locations = Location.objects.without_tree_fields().order_by().filter(devices__in=devices).only("pk").distinct()
+
+        if devices.count() > 0:
+            # Limit VRF choices to only VRFs available on all parent devices
+            vrf_choices = VRF.objects.order_by().all()
+            for parent_device in devices:
+                vrf_choices = vrf_choices.filter(devices=parent_device)
```
Comment (gsnider2195) [inline nautobot/dcim/forms.py:2438]: The performance of this query concerns me. Each additional distinct device adds another inner join. I tested this on the demo dataset against 10 interfaces on 10 distinct devices and confirmed that behavior:

```sql
SELECT "ipam_vrf"."id",
       "ipam_vrf"."name"
  FROM "ipam_vrf"
 INNER JOIN "ipam_vrfdeviceassignment"
    ON ("ipam_vrf"."id" = "ipam_vrfdeviceassignment"."vrf_id")
 INNER JOIN "ipam_vrfdeviceassignment" T4
    ON ("ipam_vrf"."id" = T4."vrf_id")
 INNER JOIN "ipam_vrfdeviceassignment" T6
    ON ("ipam_vrf"."id" = T6."vrf_id")
 INNER JOIN "ipam_vrfdeviceassignment" T8
    ON ("ipam_vrf"."id" = T8."vrf_id")
 INNER JOIN "ipam_vrfdeviceassignment" T10
    ON ("ipam_vrf"."id" = T10."vrf_id")
 INNER JOIN "ipam_vrfdeviceassignment" T12
    ON ("ipam_vrf"."id" = T12."vrf_id")
 INNER JOIN "ipam_vrfdeviceassignment" T14
    ON ("ipam_vrf"."id" = T14."vrf_id")
 INNER JOIN "ipam_vrfdeviceassignment" T16
    ON ("ipam_vrf"."id" = T16."vrf_id")
 INNER JOIN "ipam_vrfdeviceassignment" T18
    ON ("ipam_vrf"."id" = T18."vrf_id")
 INNER JOIN "ipam_vrfdeviceassignment" T20
    ON ("ipam_vrf"."id" = T20."vrf_id")
 WHERE ("ipam_vrfdeviceassignment"."device_id" = '37a938b0-bd5a-4c25-9d98-a51c75d15ba9'::uuid AND T4."device_id" = '40620024-c9b3-4e06-876f-009d51a0c1f6'::uuid AND T6."device_id" = '7042ae5f-1802-4597-917b-f93570fca692'::uuid AND T8."device_id" = '876cdcd6-21fa-4717-95aa-b3f73318a7ac'::uuid AND T10."device_id" = '9e43ed6f-0f4c-4e72-8726-037bde8d9782'::uuid AND T12."device_id" = 'b1fff0e3-8fdc-4b82-ba9f-1646124c7d09'::uuid AND T14."device_id" = 'ba16aaa7-22a7-41fa-9a4d-d132a68c7643'::uuid AND T16."device_id" = 'bb904ac6-752a-4fb1-bff4-d02b375dd707'::uuid AND T18."device_id" = 'bf9328a0-170e-4d7a-a1db-d4061053703b'::uuid AND T20."device_id" = 'eaeca318-23f3-4ef0-805c-fceb9d23a0dd'::uuid)
```

However, the SQL for this page load still completed within 90ms and the page took less than a second to load. I think it's ok to leave?

Comment (gsnider2195) [reply nautobot/dcim/forms.py:2438]: Would it be more appropriate to put an upper limit on the number of devices or perhaps just do this calculation in python? 50 devices returned in 500ms. I gave up after 4 minutes when trying to do 351 devices.

Comment (gsnider2195) [reply nautobot/dcim/forms.py:2438]: Another option is to add all VRFs and let the form validation fail if they choose a VRF that's not valid for all of the interfaces.

Comment (glennmatthews) [reply nautobot/dcim/forms.py:2438]: My vote would be to keep it simple and let the user choose amongst all VRFs (possibly filtered by a "namespace" dropdown if desirable?)

File: nautobot/dcim/forms.py [hunk 2]
```diff
@@ -2414,42 +2413,42 @@ class Meta:
...
+        if "pk" not in self.initial:
+            return
+
+        interfaces = (
+            Interface.objects.filter(pk__in=self.initial["pk"])
+            .select_related("device__location")
+            .only("id", "device", "device__location")
+        )
+        devices = Device.objects.filter(interfaces__in=interfaces).only("pk").distinct()
+        locations = Location.objects.without_tree_fields().order_by().filter(devices__in=devices).only("pk").distinct()
```
Comment (glennmatthews) [inline nautobot/dcim/forms.py:2431]: Don't these also become a performance concern when bulk-editing a large set of interfaces?

Comment (gsnider2195) [reply nautobot/dcim/forms.py:2431]: The Device query is a single join, the locations query is a single join with a subquery join. They both seem to be pretty fast for me and I've also constructed them to be as performant as possible.

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**